src/Simplify.elm

```diff
@@ -1545,83 +1545,52 @@ expressionVisitorHelp (Node expressionRange expression) context =
             }
     in
     case expression of
-        --------------------
-        -- FUNCTION CALLS --
-        --------------------
-        Expression.Application ((Node fnRange (Expression.FunctionOrValue _ fnName)) :: firstArg :: restOfArguments) ->
-            case
-                ModuleNameLookupTable.moduleNameAt context.lookupTable fnRange
-                    |> Maybe.andThen (\moduleName -> Dict.get ( moduleName, fnName ) functionCallChecks)
-            of
-                Just checkFn ->
-                    onlyErrors
-                        (checkFn
-                            (toCheckInfo
-                                { fnRange = fnRange
-                                , firstArg = firstArg
-                                , argsAfterFirst = restOfArguments
-                                , usingRightPizza = False
-                                }
-                            )
-                        )
-
-                Nothing ->
-                    onlyErrors []
-
-        -------------------
-        -- IF EXPRESSION --
-        -------------------
-        Expression.IfBlock condition trueBranch falseBranch ->
-            ifChecks
-                context
-                expressionRange
-                { condition = condition
-                , trueBranch = trueBranch
-                , falseBranch = falseBranch
-                }
-
-        -------------------------------
-        --  APPLIED LAMBDA FUNCTIONS --
-        -------------------------------
-        Expression.Application ((Node _ (Expression.ParenthesizedExpression (Node lambdaRange (Expression.LambdaExpression lambda)))) :: firstArgument :: _) ->
-            onlyErrors
-                (appliedLambdaChecks
-                    { lambdaRange = lambdaRange
-                    , lambda = lambda
-                    , firstArgument = firstArgument
-                    }
-                )
+        -----------------
+        -- APPLICATION --
+        -----------------
+        Expression.Application (applied :: applicationFirstArg :: applicationArgsAfterFirst) ->
+            case ( applied, ( applicationFirstArg, applicationArgsAfterFirst ) ) of
```

Comment:
We could pattern match on a triplet here instead of nested tuples. That said, I don't think you need to pattern match on `applicationFirstArg`, in this case it's only used to give it a different name.

And I'm even unsure as to whether we need to pattern match on `applicationArgsAfterFirst`, I'd probably do a new case expression in the `PrefixOperator` case, that way this case expression is only on `applied` and is a bit simpler. But feel free to do what you prefer for this one :+1: